plugins/cost-insights/src/components/CostOverviewCard/CostOverviewByProductChart.tsx

```diff
@@ -104,36 +109,48 @@ export const CostOverviewByProductChart = ({
 
   const chartData: Record<string, number>[] = Object.keys(productsByDate).map(
     date => {
+      const costsForDate = Object.keys(productsByDate[date]).reduce(
+        (dateCosts, product) => {
+          // Group costs for products that belong to 'Other' in the chart.
+          const cost = productsByDate[date][product];
+          const productCost = otherProducts.includes(product)
+            ? { Other: (dateCosts.Other || 0) + cost }
+            : { [product]: cost };
+          return { ...dateCosts, ...productCost };
+        },
+        {} as Record<string, number>,
+      );
       return {
-        ...productsByDate[date],
+        ...costsForDate,
         date: Date.parse(date),
       };
     },
   );
 
+  const sortedProducts = costsByProduct.sort(
+    (a, b) => aggregationSum(a.aggregation) - aggregationSum(b.aggregation),
+  );
+
   const renderAreas = () => {
-    const productGroupNames = new Set(
-      Object.values(productsByDate)
-        .map(d => Object.keys(d))
-        .flat(),
-    );
-    const sortedProducts = costsByProduct
+    const separatedProducts = sortedProducts
       // Check that product is a separate group and hasn't been added to 'Other'
       .filter(
-        product => product.id !== 'Other' && productGroupNames.has(product.id),
-      )
-      .sort(
-        (a, b) => aggregationSum(a.aggregation) - aggregationSum(b.aggregation),
+        product =>
+          product.id !== 'Other' && !otherProducts.includes(product.id),
       )
       .map(product => product.id);
     // Keep 'Other' category at the bottom of the stack
-    return ['Other', ...sortedProducts].map((product, i) => (
+    return ['Other', ...separatedProducts].map((product, i) => (
       <Area
         dataKey={product}
         isAnimationActive={false}
         stackId="1"
-        stroke={theme.palette.dataViz[sortedProducts.length - i]}
-        fill={theme.palette.dataViz[sortedProducts.length - i]}
+        stroke={theme.palette.dataViz[separatedProducts.length - i]}
+        fill={theme.palette.dataViz[separatedProducts.length - i]}
+        onClick={() => setExpanded(true)}
```

Comment:
Hm, been thinking about this a lot and not sure how I feel about the way we expand. Currently, the user would have to click on the 'Other' area on the chart, which would trigger the tooltips to now show all products. But this isn't very easy to convey to the user, but also there's not many mechanisms we can use to expand the tooltip 🤔

Reply:
I think it might be functionally easier to switch to not collapsing `Other` in the graph on click. That way there's less tooltip shenanigans. The hiccup there is that we'd have to modulo the data viz colors so they cycle through again with many products, but I sorta think that's okay (at least they are ordered).

In that scenario, I think the `[ ] Click to expand` tooltip visual from SKU breakdowns makes a bit more sense. We could also switch it to `[ ] Click to collapse` but I'm not sure it's necessary.